gulp-settings/aws-lambda.js

```diff
@@ -88,39 +88,103 @@ gulp.task('upload', () => {
 });
 
 const lambdaUploader = (lambdaFolder) => {
+    const packageJSON = require(path.join(LAMBDA_ABOSULTE_DIR, lambdaFolder, 'package.json'));
+    const lambdaFunctionName = changeCase.camelCase(packageJSON.name);
+    const lambdaFunctionDescription = packageJSON.description;
     const awsLambda = new AWS.Lambda();
-    const lambda = changeCase.camelCase(require(path.join(LAMBDA_ABOSULTE_DIR, lambdaFolder, 'package.json')).name);
+
+    return getLambdaFunction(awsLambda, lambdaFunctionName)
+        .then(() => updateLambdaFunction(awsLambda, lambdaFolder, lambdaFunctionName))
+        .catch(() => createLambdaFunction(awsLambda, lambdaFolder, lambdaFunctionName, lambdaFunctionDescription));
+};
+
+const getLambdaFunction = (awsLambda, lambdaFunctionName) => {
+    gutil.log(`Checking if lambda function: ${lambdaFunctionName} exists`);
 
     return new Promise((resolve, reject) => {
-        awsLambda.getFunction({ FunctionName: lambda }, (err) => {
+        awsLambda.getFunction({ FunctionName: lambdaFunctionName }, (err) => {
             if (err) {
                 if (err.statusCode === 404) {
-                    gutil.log(`Unable to find lambda function: ${lambda}`);
-                    gutil.log('Verify the lambda function name and AWS region are correct');
+                    gutil.log(`Unable to find lambda function: ${lambdaFunctionName}, creating it now...`);
                 } else {
                     gutil.log('AWS API request failed. Check your AWS credentials and permission');
                 }
 
                 reject(err);
-            } else {
-                fs.readFile(path.join(LAMBDA_ABOSULTE_DIR, lambdaFolder, 'build.zip'), (err, data) => {
-                    const params = {
-                        FunctionName: lambda,
-                        ZipFile: data
-                    };
-
-                    awsLambda.updateFunctionCode(params, (err) => {
-                        if (err) {
-                            gutil.log(err);
-                            gutil.log('Package upload failed. Check your iam:PassRole permission');
-                            reject(err);
-                        }
-
-                        gutil.log(`Successfully uploaded: ${chalk.green(lambda)}`);
-                        resolve('success');
-                    });
+            }
+
+            resolve('Success');
+        });
+    });
+};
+
+const updateLambdaFunction = (awsLambda, lambdaFolder, lambdaFunctionName) => {
+    gutil.log(`Updating lambda function: ${lambdaFunctionName}...`);
+
+    return new Promise((resolve, reject) => {
+        readLambdaBuild(lambdaFolder, lambdaFunctionName)
+            .then((fileData) => {
+                awsLambda.updateFunctionCode({
+                    FunctionName: lambdaFunctionName,
+                    ZipFile: fileData
+                }, (err) => {
```

Comment:
For the better readability I think it would be better to extract the error handling:
```
readLambdaBuild(lambdaFolder, lambdaFunctionName)
    .then(fileData => {...})
    .catch(err => {...})
```